core/src/main/java/com/valtech/aem/saas/core/indexing/DefaultIndexUpdateService.java

```diff
@@ -0,0 +1,154 @@
+            .setContentEncoding(StandardCharsets.UTF_8.name())
+            .build())
+        .build();
+    return searchRequestExecutorService.execute(searchRequest)
+        .flatMap(response -> response.get(new DefaultIndexUpdateDataExtractionStrategy()));
+  }
+
+  @Override
+  public Optional<IndexUpdateResponse> indexContent(@NonNull String client,
+      @NonNull IndexContentPayload indexContentPayload) {
+    validateInputClient(client);
+    SearchRequest searchRequest = SearchRequestPost.builder()
+        .uri(getRequestUri(client, configuration.indexUpdateService_apiPushContentAction()))
+        .httpEntity(EntityBuilder.create().setText(new Gson().toJson(indexContentPayload))
+            .setContentType(ContentType.APPLICATION_JSON)
+            .setContentEncoding(StandardCharsets.UTF_8.name())
+            .build())
+        .build();
+
+    return searchRequestExecutorService.execute(searchRequest)
+        .flatMap(response -> response.get(new DefaultIndexUpdateDataExtractionStrategy()));
+  }
+
+  private String getRequestUri(String client, String action) {
+    return String.format("%s%s%s%s%s", searchServiceConnectionConfigurationService.getBaseUrl(),
+        configuration.indexUpdateService_apiBasePath(), client,
+        configuration.indexUpdateService_apiVersionPath(), action);
+  }
+
+  private void validateInput(String client, String url, String repositoryPath) {
+    validateInputClient(client);
+    if (StringUtils.isBlank(url)) {
+      throw new IllegalArgumentException("Please pass a url of a content that should be indexed.");
+    }
+    if (StringUtils.isBlank(repositoryPath)) {
+      throw new IllegalArgumentException("Please pass a repository path regex value.");
+    }
+  }
+
+  private void validateInputClient(String client) {
+    if (StringUtils.isBlank(client)) {
+      throw new IllegalArgumentException("Please configure client in context aware configuration.");
+    }
+  }
+
+  @ObjectClassDefinition(name = "Search as a Service - Index Update Service Configuration",
+      description = "Index Update Api specific details.")
+  public @interface Configuration {
+
+    String DEFAULT_API_INDEX_TRIGGER_ACTION = "/index/trigger";
+    String DEFAULT_API_PUSH_CONTENT_ACTION = "/content";
+    String DEFAULT_API_BASE_PATH = "/admin";
+    String DEFAULT_API_VERSION_PATH = "/api/v3";
+
+    @AttributeDefinition(name = "Api base path",
+        description = "Api base path",
+        type = AttributeType.STRING)
+    String indexUpdateService_apiBasePath() default DEFAULT_API_BASE_PATH;
+
+    @AttributeDefinition(name = "Api version path",
+        description = "Api base path",
+        type = AttributeType.STRING)
+    String indexUpdateService_apiVersionPath() default DEFAULT_API_VERSION_PATH;
+
+    @AttributeDefinition(name = "Api index trigger action",
+        description = "What kind of action should be defined",
```

Comment:
is this description correct?

Reply:
I guess it is not the best description :). I'll try to improve it